src/TagField.php

```diff
@@ -276,8 +276,12 @@ protected function getOptions($onlySelected = false)
         // Convert an array of values into a datalist of options
         if (!$values instanceof SS_List) {
             if (is_array($values) && !empty($values)) {
-                $values = DataList::create($dataClass)
-                    ->filter($this->getTitleField(), $values);
+                if (is_subclass_of($source, DataList::class) || is_a($source, DataList::class)) {
```

Comment:
@obj63mc `is_a($source, DataList::class)` Checks if `$source` is of class `DataList` **OR** has `DataList` as one of its parents. So I think `is_subclass_of($source, DataList::class)` can be omitted.